### PR TITLE
Fix Orca pytorch metrics MAE and MSE

### PR DESCRIPTION
**Pull request by yaxinchen666**
Modify MAE and MSE to support float labels.
Remove the line making the type of `preds` the same as `target`. (reference: https://pytorch.org/docs/stable/generated/torch.sub.html#torch.sub
https://pytorch.org/docs/stable/tensor_attributes.html#type-promotion-doc
)
Add and run unit test.

### CODE REVIEW

File: pyzoo/test/zoo/orca/learn/test_metrics.py [hunk 1]
```diff
@@ -94,6 +94,14 @@ def test_torch_MAE():
...
     m(pred, target)
     assert m.compute() == 0.875
+    pred = torch.tensor([[1.5, 2.5], [1.0, 1.0]])
+    target = torch.tensor([[0.2, 1.1], [0.5, 1.0]])
+    m(pred, target)
+    assert abs(m.compute() - 0.85) < 1e-6 
+    pred = torch.tensor([[1.5, 2.5], [1.8, 2.0]])
+    target = torch.tensor([[0, 1], [0, 1]])
+    m(pred, target)
+    assert abs(m.compute() - 1) < 1e-6
```
Comment (hkvision) [inline pyzoo/test/zoo/orca/learn/test_metrics.py:104]: Why there is fault tolerance? Is there any issue that m.compute() == a specific value?

File: pyzoo/test/zoo/orca/learn/test_metrics.py [hunk 2]
```diff
@@ -94,6 +94,14 @@ def test_torch_MAE():
     target = torch.tensor([[0, 1], [0, 1]])
     m(pred, target)
     assert m.compute() == 0.875
+    pred = torch.tensor([[1.5, 2.5], [1.0, 1.0]])
```
Comment (hkvision) [inline pyzoo/test/zoo/orca/learn/test_metrics.py:97]: Maybe change one unit test to tensors with different dimensions?

File: pyzoo/test/zoo/orca/learn/test_metrics.py (hunk 1 above)
Comment (yaxinchen666) [reply pyzoo/test/zoo/orca/learn/test_metrics.py]: I tried `==` at the beginning, and one of the unit test failed:
FAILED test_metrics.py::test_torch_MAE - assert tensor(0.8500) == 0.85
I think it is a float point issue:
https://github.com/pytorch/pytorch/issues/15900

**Comment by hkvision** [comment]
@leonardozcm Take a look.

**Review by hkvision — APPROVED** [review]

**Comment by yaxinchen666** [comment]
Jenkins:
http://10.239.47.210:18888/job/ZOO-PR-Validation/5228/
Only ZOO-PR-Python-Spark-2.4-py36 failed. Reran ZOO-PR-Python-Spark-2.4:
http://10.239.47.210:18888/job/ZOO-PR-Validation-Spark-2.4/3596/
Some failed but ZOO-PR-Python-Spark-2.4-py36 passed. Errors are unrelated.

File: pyzoo/test/zoo/orca/learn/test_metrics.py [hunk 3]
```diff
@@ -94,6 +94,14 @@ def test_torch_MAE():
...
+    target = torch.tensor([[0.2, 1.1], [0.5, 1.0]])
+    m(pred, target)
+    assert abs(m.compute() - 0.85) < 1e-7   # add float tolerance for floating point precision
```
Comment (hkvision) [inline pyzoo/test/zoo/orca/learn/test_metrics.py:100]: float -> fault

Comment (yaxinchen666) [reply pyzoo/test/zoo/orca/learn/test_metrics.py]: 2333

**Comment by leonardozcm** [comment]
> @leonardozcm Take a look.

LGTM
